lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,254 @@
+  var output = <Token>[];
+  var tokenID = 0;
+  var firstInLine = true;
+
+  for (var token in tokens) {
+    // Ignore new line tokens for now.
+    // If accuracy of detection is low apply
+    // Guideline 2.1.4: Text that can be omitted from license.
+    if (_newLineRegex.hasMatch(token.value)) {
+      firstInLine = true;
+      continue;
+    }
+
+    // Ignores list items.
+    if (firstInLine && _isListItem(token.value)) {
+      continue;
+    }
+
+    firstInLine = false;
+
+    final text = _cleanToken(token.value);
+
+    output.add(Token(text, tokenID++, token.span));
+  }
+
+  return output;
+}
+
+/// Normalizes the tokens using the approach of [google licenseClassifier]
+/// to provide better chances of matching.
+///
+/// [google licenseClassifier]  https://github.com/google/licenseclassifier/blob/bb04aff29e72e636ba260ec61150c6e15f111d7e/v2/tokenizer.go#L85
+String _cleanToken(String tok) {
+  if (tok.startsWith(RegExp(r'[^a-z]'))) {
+    if (tok.startsWith(RegExp(r'\d'))) {
+      var text = tok.replaceAll(RegExp(r'[^\d\.-]'), '');
+
+      if (text.endsWith('.')) {
+        text = text.substring(0, text.length - 1);
+      }
+
+      return text;
+    }
+  }
+
+  var text = tok.replaceAll(RegExp(r'[^a-z]'), '');
+
+  // If this is a varietal word normalize it according to
+  // Guideline 8.1.1: Legally equal words must be treated same.
+  text = _varietalWords[text] ?? text;
+  return text;
+}
+
+bool _isListItem(String token) {
+  final end = token[token.length - 1];
+  final start = token.substring(0, token.length - 1);
+
+  if ((end == '.' || end == ')') && _headers.contains(start)) {
+    return true;
+  }
+
+  if (_numberHeaderRe.hasMatch(token)) {
+    return true;
+  }
+
+  return false;
+}
+
+String _normalizeWord(String text) {
+  // Guideline 5: Equivalent Punctuation marks.
+  _equivalentPunctuationMarks.forEach((reg, value) {
+    text = text.replaceAll(reg, value);
+  });
+
+  return text;
+}
+
+final _headers = HashSet.from(
+    'q w e r t y u i o p a s d f g h j k l z x c v b n m i ii iii iv vi vii ix xi xii xiii xiv xv'
+        .split(' '));
+
+final _numberHeaderRe = RegExp(r'^\d{1,2}(\.\d{1,2})*[\.)]$');
+
+final _horizontalWhiteSpaceRegex = RegExp(r'[^\S\r\n]+');
+
+final Map<RegExp, String> _equivalentPunctuationMarks = {
+  // Guideline 5.1.2: All variants of dashes considered equivalent.
+  RegExp(r'[-֊־᠆‐‑–—﹘﹣－‒⁃⁻⎯─⏤]'): '-',
+
+  // Guideline Guideline 5.1.3: All variants of quotations considered equivalent.
+  RegExp(r'[“‟”’"‘‛❛❜〝〞«»‹›❝❞]'): "'",
+
+  // Guideline 9.1.1: “©”, “(c)”, or “Copyright” should be considered equivalent and interchangeable.
+  RegExp(r'©'): '(c)'
+};
+
+final _wordRegex = RegExp(r'[\w\d][^\s]*');
+final _newLineRegex = RegExp(r'(?:\n|\r\n|\r|\u0085)');
+
+/// Words considered legally equivalent according [SPDX corpus][1].
+///
+/// [1]: https://github.com/spdx/license-list-XML/blob/master/equivalentwords.txt
+final _varietalWords = {
```

Comment:
I would use the same term as the referenced file:
```suggestion
final _equivalentWords = {
```